core/src/main/java/org/apache/calcite/sql/SqlWindowTableFunction.java

```diff
@@ -244,5 +253,53 @@ void validateColumnNames(SqlValidator validator,
         }
       });
     }
+
+    /**
+     * Checks whether input table satisfy requirement of input semantics.
+     * Based on SQL standard 2016 Polymorphic Table Functions:
+     * the input with row semantics may not be partitioned.
+     * the input with set semantics may be partitioned on one or more columns.
+     *
+     * @param callBinding The call binding
+     * @return true if validation passes
+     */
+    boolean checkInputTableSemantic(SqlCallBinding callBinding) {
+      boolean isValid;
```

Comment:
Do we really need this ? How can people write a tumble or hop window function with partition by ?

Reply:
Hi, Danny, I add this restrict because PTF SQL standard.
Only input tables with set semantics may be partitioned on columns while input table with row semantics may not be partitioned or ordered.
![image](https://user-images.githubusercontent.com/1525333/134791801-2c1a21ad-afdd-4024-a48a-a21a1e14ec03.png)

Based on this rule, tumbling window and Hop window function could not be with partition by because their input table has row semantics.